lib/semian/net_http.rb

```diff
@@ -0,0 +1,90 @@
+require 'semian'
+require 'semian/adapter'
+
+module Net
+  ::Net::ProtocolError.include(::Semian::AdapterError)
+
+  class SemianError < ::Net::ProtocolError
+    def initialize(semian_identifier, *args)
+      super(*args)
+      @semian_identifier = semian_identifier
+    end
+  end
+
+  Net::ResourceBusyError = Class.new(SemianError)
+  Net::CircuitOpenError = Class.new(SemianError)
```

Comment:
Can you just do `ResourceBusyError` instead of prefixing with the namespace you're already in?